kedro/framework/cli/starters.py

```diff
@@ -149,11 +217,32 @@ def starter():
 @starter.command("list")
 def list_starters():
     """List all official project starters available."""
-    repo_url = _STARTERS_REPO.replace("git+", "").replace(".git", "/tree/main/{alias}")
-    output = [
-        {alias: repo_url.format(alias=alias)} for alias in sorted(_STARTER_ALIASES)
-    ]
-    click.echo(yaml.safe_dump(output))
+    starters_aliases = _get_starters_aliases()
+    # we "reverse" the dictionnary so that the origin key in starter config
+    # {name: {template_path; ..., origin: ...}} became the first key:
+    # {kedro: ["astro-iris":{...}], "my-plugin":["my-starter":{...}]}
+    starters_by_origin = {}
+    for starter_name, starter_config in starters_aliases.items():
+        starter_raw_config = {
+            starter_name: {k: v for k, v in starter_config.items() if k != "origin"}
+        }
+        if starter_config["origin"] in starters_by_origin:
+            starters_by_origin[starter_config["origin"]].append(starter_raw_config)
+        else:
+            starters_by_origin[starter_config["origin"]] = [starter_raw_config]
+    for origin, starter_aliases in starters_by_origin.items():
+        click.secho(f"Starters from '{origin}'\n", fg="yellow")
+        output = []
+        for starter_alias in starter_aliases:
+            for starter_name, starter_config in starter_alias.items():
+                starter_name_displayed = (
+                    f"{starter_name} "
+                    f"(template_path={starter_config.get('template_path')}, "
+                    f"directory={starter_config.get('directory')}, "
+                    f"checkout={starter_config.get('checkout')})"
+                )
+                output.append(starter_name_displayed)
+        click.echo(yaml.safe_dump(output))
```

Comment:
There's a few possible simplifications here. I think we can [rely on Python dictionary ordering](https://stackoverflow.com/questions/1867861/how-to-keep-keys-values-in-same-order-as-declared), so you can do it without needing to restructure the dictionary at all:
```
    last_origin = "kedro"
    click.secho(f"Built-in starters\n", fg="yellow")

    for starter_name, starter_config in starters_aliases.items():
        origin = starter_config.pop("origin")
        if origin != last_origin:
            click.secho(f"\nStarters from {origin}\n", fg="yellow")
            last_origin = origin
        click.echo(f"{starter_name}: {starter_config}")
```
If you do want to restructure the dictionary then it's probably possible using something like [this](https://www.geeksforgeeks.org/group-list-of-dictionary-data-by-particular-key-in-python/).

The above renders as follows:
![image](https://user-images.githubusercontent.com/49395058/155129113-72ca933d-257e-475c-b0ad-813a9137f0c4.png)

... which is not super beautiful I will admit 😅 Could probably be made nicer by using `yaml.dump(default_flow_style=True)` instead of just printing the raw dictionary. But I do like only giving `directory` and `checkout` when they're explicitly defined.

Reply:
@AntonyMilneQB I thought dictionnary ordering was only from ``python>=3.7`` and forward and I exclitly avoid relying on dict ordering because I know you still support ``python=3.6`` :) But I was wrong so glad to make it more readable.

@ lorenabalan I think `pop()` should be fine because the ``_get_starter_aliases()`` is called at the beginning of the function each time, and it does itself copy the _STARTERS_ALIASES constant. It would be safer to explictly deepcopy the dict in case of future changes though, do you prefer to enforce it?

I am still experimenting to see if we can have something more readable.